src/test/java/ro/secur/auth/service/UserServiceTest.java

```diff
@@ -0,0 +1,189 @@
+    }
+
+    void mockUserRepoReturnsNull() {
+        when(userRepository.findByUserName(any(String.class))).thenReturn(null);
+    }
+
+
+    @Test
+    void whenLoadUserByUsername_returnUser() {
+
+        mockUserRepoReturnsUser("test", "pass", Role.USER);
+
+        User expectedUser = new User("test", "pass",
+                Collections.singletonList(new SimpleGrantedAuthority(Role.USER.name())));
+        User actualUser = (User) userService.loadUserByUsername("test");
+
+        Assertions.assertEquals(expectedUser, actualUser);
+    }
+
+
+    @Test
+    void whenLoadUserByUsername_throwException() {
+
+        mockUserRepoReturnsNull();
+
+        UserNotFoundException exception = assertThrows(UserNotFoundException.class,
+                () -> userService.loadUserByUsername("test"));
+
+        String expectedMessage = "Entity not found in the database for user: test";
+        String actualMessage = exception.getMessage();
+
+        Assertions.assertEquals(expectedMessage, actualMessage);
+    }
+
+
+    @Test
+    void getAllUsers() {
+        //given
+        UserEntity user = UserEntity.builder()
+                .id(3L)
+                .userName("TOM-MAC-BILL-BOB-CONSTANTIN")
+                .password("marcel")
+                .build();
+        List<UserEntity> userEntities = Lists.newArrayList(user);
+
+        //when
+        when(userRepository.findAll()).thenReturn(userEntities);
+
+        List<UserEntity> receivedEntities = (List<UserEntity>) userRepository.findAll();
+
+        //then
+        assertEquals(receivedEntities.size(), 1);
+        verify(userRepository, times(1)).findAll();
+    }
+
+
+    @Test
+    void changePassword() {
+        //given
+        UserEntity userEntity = UserEntity.builder()
+                .id(1L)
+                .userName("BruceLee")
+                .password("kongfuuuuu")
+                .build();
+        //when
+        when(userRepository.findByUserName(any(String.class))).thenReturn(userEntity);
+
+        //then
+        assertEquals(userEntity.getPassword(), "kongfuuuuu");
+    }
+
+
+    @Test
+    void checkPassword_thenThrowInvalidPasswordEx() {
+        //given
+        ChangePasswordRequest request = ChangePasswordRequest.builder()
+                .username("lungu")
+                .password("JohnnyBravo")
+                .newPassword("JohnnyBravo")
+                .build();
+
+        UserEntity userEntity = UserEntity.builder()
+                .id(1L)
+                .userName("BruceLee")
+                .password("kongfuuuuu")
+                .build();
+
+        //when
+        when(userRepository.findByUserName(any(String.class))).thenReturn(userEntity);
+        when(passwordConfiguration.verifyHash(anyString(), anyString())).thenReturn(false);
+
+        //then
+        assertThrows(InvalidPasswordException.class, () -> userService.changePassword(request));
+
+    }
+
+
+    @Test
+    void checkPassword_thenThrowMismatchEx() {
```

Comment:
name test methods similar with the method you want to test (e.g. changePassword, not checkPassword)

Reply:
ok